src/ast/stmt/SwitchStmt.php

```diff
@@ -64,4 +65,22 @@ public function injectScope(&$parent_scope)
             $case->injectScope($parent_scope);
         }
     }
+
+    public function runTypeChecker()
+    {
+        $value_type = $this->value->getType();
+        foreach ($this->cases as $case) {
+            if (!$case->is_else) {
+                $case_type = $case->value->getType();
+                if ($case_type->code !== $value_type->code) {
+                    // TODO: how to get the case expression value? 
```

Comment:
@haskellcamargo I did try to do `$case->value`, although it was throwing an en error like "The type StringExpr cannot be converted to string", and then I changed it to `$case->value->value`. It makes sense once it is a value inside a value. 
Although, that does not work for objects. The parser accepts objects as cases for example:
```Swift
let y :- "blah"
let x :- 1
switch x
    case 0 /* stmt */
    case y /* stmt */
end
```
In the case above, `y` has the type of `NameExpr` and the `value` field is not present for that type. It is called `name` instead.
My question is: shouldn't we have a `toString`-like method on `Quack`?

Other than that, I'm running the `runTypeChecker` already for each case here: [SwitchStmt.php#L83](https://github.com/luizperes/quack/blob/b2d31fee13d3cce68dc9760d7dc390e4fbf542ab/src/ast/stmt/SwitchStmt.php#L83)